src/main/java/com/mide/gangsaeng/board/BoardServiceImpl.java

```diff
@@ -20,23 +22,30 @@ public class BoardServiceImpl implements BoardService {
     public static final int MAX_VALUE_OF_PAGE_SIZE = 30;
     private final BoardRepository boardRepository;
     private final BannedWordService bannedWordService;
+    private final BoardMapper boardMapper;
 
     @Autowired
     public BoardServiceImpl(@Qualifier("cacheableBoardRepository") BoardRepository boardRepository,
-                            BannedWordService bannedWordService) {
+                            BannedWordService bannedWordService,
+                            BoardMapper boardMapper) {
         this.boardRepository = boardRepository;
         this.bannedWordService = bannedWordService;
+        this.boardMapper = boardMapper;
     }
 
     @Override
     public void write(BoardRequest request) {
         bannedWordService.validateBannedWords(request.getTitle() + " " +request.getContent());
-        boardRepository.write(request);
+
+        Board board = boardMapper.boardRequestToBoard(request);
+        boardRepository.write(board);
     }
 
     @Override
     public BoardResponse read(long id) {
         Board board = boardRepository.read(id);
+        Optional.ofNullable(board)
+                .orElseThrow(() -> new BoardNotFoundException(ErrorCode.BOARD_NOT_FOUND));
```

Comment:
Why did you use Optional here?
We can just check null directly

Reply:
I thought I could express null and exception handling with concise code...